act/io/noaapsl.py

```diff
@@ -16,151 +16,60 @@
-    site_loc = df.iloc[idx[0][0]]
-    site_list = site_loc.str.split(r'\s{2}').tolist()
-    site = site_list[0][0].strip()
-
-    obj_low.attrs['site_identifier'] = site
-    obj_hi.attrs['site_identifier'] = site
-
-    # Adding data type and revision number line 2.
-    rev = df.loc[idx[0][0] + 1]
-    rev_list = rev.str.split(r'\s{3}').tolist()
-    rev_array = np.array(rev_list[0])
-
-    obj_low.attrs['data_type'] = rev_array[0].strip()
-    obj_hi.attrs['data_type'] = rev_array[0].strip()
-    obj_low.attrs['revision_number'] = rev_array[1].strip()
-    obj_hi.attrs['revision_number'] = rev_array[1].strip()
-
-    # Adding coordinate attributes line 3.
-    coords = df.loc[idx[0][0] + 2]
-    coords_list = coords.str.split(r'\s{2,}').tolist()
-    coords_list[0].remove('')
-    coords_array = np.array(coords_list[0], dtype='float32')
-
-    obj_low.attrs['latitude'] = np.array([coords_array[0]])
-    obj_hi.attrs['latitude'] = np.array([coords_array[0]])
-    obj_low.attrs['longitude'] = np.array([coords_array[1]])
-    obj_hi.attrs['longitude'] = np.array([coords_array[1]])
-    obj_low.attrs['altitude'] = np.array([coords_array[2]])
-    obj_hi.attrs['altitude'] = np.array([coords_array[2]])
-
-    # Adding azimuth and elevation line 9
-    az_el = df.loc[idx[0][0] + 8]
-    az_el_list = az_el.str.split(r'\s{2,}').tolist()
-    az_el_list[0].remove('')
-    az_el_array = np.array(az_el_list[0])
-    az = []
-    el = []
-    for i in az_el_array:
-        sep = i.split()
-        az.append(sep[0])
-        el.append(sep[1])
-    az_array = np.array(az, dtype='float32')
-    el_array = np.array(el, dtype='float32')
-
-    obj_low.attrs['azimuth'] = az_array
-    obj_hi.attrs['azimuth'] = az_array
-    obj_low.attrs['elevation'] = el_array
-    obj_hi.attrs['elevation'] = el_array
+    # Open the file, read in the lines as a list, and return that list
+    file = fsspec.open(filepath).open()
+    lines = file.readlines()
+    lines = [x.decode().rstrip()[:] for x in lines][1:]
 
-    if transpose:
-        obj_low = obj_low.transpose()
-        obj_hi = obj_hi.transpose()
+    # Separate sections based on the $ separator in the file
+    sections_of_file = (list(g) for _, g in groupby(lines, key='$'.__ne__))
 
-    return obj_low, obj_hi
+    # Count how many lines need to be skipped when reading into pandas
+    start_line = 0
+    list_of_datasets = []
+    for section in sections_of_file:
+        if section[0] != '$':
+            list_of_datasets.append(
+                _parse_psl_wind_lines(
+                    filepath, section, line_offset=start_line)
+            )
+        start_line += len(section)
+
+    # Return two datasets for each mode and the merge of datasets of the
+    # same mode.
+    mode_one = xr.concat(
+        list_of_datasets[0::2], dim='time')
+    mode_two = xr.concat(
+        list_of_datasets[1::2], dim='time')
+    if transpose:
+        mode_one = mode_one.transpose('HT', 'time')
```

Comment:
Was there a reason for going with HT vs height?

Reply:
@AdamTheisen looks like originally in the data it was HT and we renamed it in the original code, so was keeping it HT to represent the original variable name.

Reply:
Yeah, this is consistent with how we dealt with a similar file with temperature data

Reply:
Sounds good!